control-plane/common/controlplane/client.go

```diff
@@ -4,38 +4,48 @@
 package controlplane
 
 import (
+	"context"
 	"fmt"
 
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
 
 	"github.com/agntcy/slim/control-plane/common/options"
 	cpApi "github.com/agntcy/slim/control-plane/common/proto/controlplane/v1"
 )
 
 func GetClient(
 	opts *options.CommonOptions,
-) (cpApi.ControlPlaneServiceClient, error) {
+	ctx context.Context,
+) (cpApi.ControlPlaneServiceClient, context.Context, error) {
 	var creds credentials.TransportCredentials
 	if opts.TLSInsecure {
 		creds = insecure.NewCredentials()
 	} else if opts.TLSCAFile != "" {
 		c, err := credentials.NewClientTLSFromFile(opts.TLSCAFile, "")
 		if err != nil {
-			return nil, fmt.Errorf("loading CA file %q: %w", opts.TLSCAFile, err)
+			return nil, ctx, fmt.Errorf("loading CA file %q: %w", opts.TLSCAFile, err)
 		}
 		creds = c
 	}
 
+	if opts.BasicAuthKey != "" {
+		md := metadata.New(map[string]string{
+			"authorization": "Basic " + opts.BasicAuthKey,
+		})
+		ctx = metadata.NewOutgoingContext(ctx, md)
+	}
```

Comment:
Probably not for this PR, but we could think of using [client interceptors](https://github.com/grpc-ecosystem/go-grpc-middleware/tree/main/interceptors/auth) to handle other stuff like rate limiting, logging etc.